src/pattern-library/components/Library.tsx

```diff
@@ -332,48 +396,91 @@ function ChangelogItem({ status, children }: LibraryChangelogItemProps) {
   );
 }
 
+type CodeContentProps =
+  | {
+      /** Code content (to be rendered with syntax highlighting) */
+      content: ComponentChildren;
+    }
+  | {
+      /**
+       * Example file to read and use as content (to be rendered with syntax
+       * highlighting)
+       */
+      exampleFile: string;
+    };
+
 export type LibraryCodeProps = {
-  /** Code content (to be rendered with syntax highlighting) */
-  content: ComponentChildren;
   /** Controls relative code font size */
   size?: 'sm' | 'md';
   /** Caption (e.g. filename, description) of code block */
   title?: ComponentChildren;
-};
+} & CodeContentProps;
+
+function isCodeWithContent(
+  props: CodeContentProps,
+): props is { content: ComponentChildren } {
+  return 'content' in props;
+}
+
+function useCodeContent(
+  props: CodeContentProps,
+): [string | undefined, Error | undefined] {
+  const [codeMarkup, setCodeMarkup] = useState<string>();
+  const [error, setError] = useState<Error>();
+  const setJSXContent = useCallback(
+    (content: ComponentChildren) => setCodeMarkup(jsxToHTML(content)),
```

Comment:
`jsxToHTML` expects a JSX expression (`ComponentChildren`), but we are passing a string here. That happens to work because `string` is a valid `ComponentChildren` and `jsxToHTML` handles strings by not touching them ... however it is rather accidental. What would make more sense is to extract out the part of `jsxToHTML` which takes code as input and highlights it, and then call that here.

While you're at it, a very easy improvement in `jsxToHTML` will be to change the language from JavaScript to TypeScript. That will fix the lack of highlighting for the `type` keyword.